### ISSUE TITLE
onUnmounted lifecycle hook not running when closing modal

### ISSUE THREAD

**Issue by j-codes-4** [lepikhinb/momentum-modal-plugin#1]
When closing a modal, `onUnmounted` is not being triggered.  `onUnmounted` will trigger on the next opening the same or a different modal, right before `onMounted` fires.  

Reproduce by adding to a modal component
```
onMounted( () => {
  console.log( 'mounted' )
} )
onUnmounted( () => {
  console.log( 'is unmounted' )
} )
```
Then open and close for a first time.  The console won't show 'is unmounted' until opening a second time.

While troubleshooting, I found the following fixed the problem.  In use-modal.ts..
```
const close = (shouldRedirect: boolean = false) => {
  show.value = false

  resetHeaders()

  // breaks animations without a timeout
  setTimeout(() => {
    component.value = false
    componentName.value = false
    vnode.value = false
  }, 1000);

  if (shouldRedirect) {
    redirect()
  }
}
```

Sorry for not making a pull request, I'm still learning and worried I would do it wrong.
Thanks for reading!

**Comment by lepikhinb** [comment]
Resolved in fecf0a0. Don't forget to call the `redirect` method on modal fade. Check the [example](https://github.com/lepikhinb/momentum-modal-plugin/blob/master/examples/Modal.vue).

Thanks!

**Comment by j-codes-4** [comment]
Thanks for the fast response and thanks for momentum-modal, it is awesome.
